pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1482,6 +1482,26 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				assert.Equal(t, "manual-resourcegroup", deprovision.Spec.Platform.Azure.ResourceGroupName)
 			},
 		},
+		{
+			name: "set ProvisionStoppedCondition when install restarts limit has reached",
```

Comment:
We should have one test where `InstallRestarts` is less than the limit, one test where `InstallRestarts` is equal to the limit, and one test where `InstallRestarts` is greater than the limit.